community/kernel/src/main/java/org/neo4j/kernel/impl/api/index/OnlineIndexUpdatesValidator.java

```diff
@@ -55,11 +58,18 @@ public OnlineIndexUpdatesValidator( NeoStore neoStore, PropertyLoader propertyLo
     }
 
     @Override
-    public ValidatedIndexUpdates validate( TransactionRepresentation transaction )
-            throws IOException
+    public ValidatedIndexUpdates validate( TransactionRepresentation transaction ) throws IOException
     {
         NodePropertyCommandsExtractor extractor = new NodePropertyCommandsExtractor();
-        transaction.accept( extractor );
+        try
+        {
+            transaction.accept( extractor );
+        }
+        catch ( IOException cause )
```

Comment:
Hmm, it looks like there is a reason. In both `RecoveryVisitor` and `TransactionCommittingResponseUnpacker` we check validity before applying to the store. Moreover, for the latter we actually append txs to the log first and  we validate them only before applying them on the store. So it looks like something is wrong here.